pcl_conversions/include/pcl_conversions/pcl_conversions.h

```diff
@@ -107,31 +110,35 @@ namespace pcl_conversions {
   /** PCLHeader <=> Header **/
 
   inline
-  void fromPCL(const pcl::PCLHeader &pcl_header, std_msgs::Header &header)
+  void fromPCL(const pcl::PCLHeader &pcl_header, std_msgs::msg::Header &header)
   {
-    fromPCL(pcl_header.stamp, header.stamp);
-    header.seq = pcl_header.seq;
+    auto time_stamp = std::make_shared<rclcpp::Time>(header.stamp);
```

Comment:
You make a shared pointer to deference it, you just need the object. If for some reason you need a pointer, use the unique_ptr

Reply:
I was thinking just to make sure it gets freed when we go out of the scope. Is there another way without using `new` and `delete` ?

Reply:
Why do you need a pointer at all? You just dereference it soon after

Reply:
No probably not. I was just thinking that if I use a smart pointer, it will automatically delete the object after the function scope is gone, no ? 

Constructing a new object would mean I need to do something like:

```
auto time_stamp = new rclcpp::Time(header.stamp);
fromPCL(pcl_header.stamp, time_stamp);
delete time_stamp;
```

But maybe that is not the way smart ptrs are intended to be used. Do you have any other recommendations on how to write this or is the code sample above an OK approach?

Reply:
You can just make a stack variable:

```
fromPCL(pcl_header.stamp, rclcpp::Time(header.stamp));
```

No smart pointers, pointers, or heap allocations required, and a single line of code :).

Reply:
that's what I was getting at. There's no reason for this to be a pointer

Reply:
Ah ok awesome, I was (and I guess still am) a bit unsure on when it's possible to allocate on the stack and on the heap, but it seems after reading up on it more that it mostly comes down to the size of the object. If it's small enough always use the stack ( ?).

Thanks for your patience, I'll fix this.

Reply:
There's a number of reasons to do one versus the other, but that is one reason, yes. 

There's a real performance downside with using heap memory so its best to not do it unless you require it. Plus if you look at what you actually ended up doing:

- create pointer
- dereference pointer to pass to a method

There wasn't anything that used it in pointer format, so there wasn't anything to be derived from the allocation

Reply:
Typically, if you are only going to use it in this function (i.e. this stack frame), you do a stack allocation.  If it needs to live beyond this function, then you do a heap allocation (either with a "raw" pointer or a smart pointer).  Size can also come into it, as on Linux the default stack size is limited by default (8MB if I remember correctly, but don't quote me).

Reply:
Not to pile on, but also if you're concerned about real-time or software certifications, heap allocations are the devil

Reply:
Thanks for all elaborate examples and explanations, really valuable!

Reply:
Stack variable worked, however couldn't get it to compile using the one-liner 
`fromPCL(pcl_header.stamp, rclcpp::Time(header.stamp));`